sound/soc/sof/sof-client.h

```diff
@@ -39,6 +41,7 @@ int sof_client_ipc_tx_message(struct sof_client_dev *cdev, void *ipc_msg,
 			      void *reply_data, size_t reply_bytes);
 int sof_client_ipc_set_get_data(struct sof_client_dev *cdev, void *ipc_msg,
 				bool set);
+struct sof_ipc4_fw_module *sof_client_ipc4_find_module_by_uuid(struct sof_client_dev *cdev, const guid_t *uuid);
 
```

Comment:
this function is only used by ipc4-probe.c, we don't need to define it here for general client.
If you want to make it useable for future,  I recommend it define in ipc4-loader.c and add anther ops in struct sof_ipc_fw_loader_ops

Reply:
I understood from earlier comments that a sof-client should not access struct snd_sof_dev directly, and if that stands a function using client API is needed.

Reply:
Yes, that's right. SOF clients _must_ not access anything from the core directly and the access should be kept minimum.